packages/taucmdr/cf/software/libelf_installation.py

```diff
@@ -45,9 +49,21 @@ class LibelfInstallation(AutotoolsInstallation):
     """Encapsulates a libelf installation."""
 
     def __init__(self, sources, target_arch, target_os, compilers):
+        if compilers[CC].unwrap().info.family is PGI or compilers[CC].unwrap().info.family is NVHPC :
+            try:
+                gnu_compilers = GNU.installation()
+            except ConfigurationError as err:
+                raise SoftwarePackageError("GNU compilers (required to build libelf) could not be found.") from err
+            compilers = compilers.modify(Host_CC=gnu_compilers[CC], Host_CXX=gnu_compilers[CXX])
         super().__init__('libelf', 'libelf', sources,
                                                  target_arch, target_os, compilers, REPOS, None, LIBRARIES, HEADERS)
 
     def configure(self, flags):
         flags.extend(['--disable-debuginfod'])
-        return super().configure(flags)
+        cc = None
+        if 'CC' in os.environ :
+            cc = os.environ['CC']
```

Comment:
@vikram8128 why is this necessary? Generally we try to really sanitize the user environment and not rely on user set environment variables. Sometimes we need to set environment variables for configure scripts and other software building but if there is a way to avoid reading CC from the environment into a variable that would be better.